packages/sample-app/src/components/app-minimal/MinimalAppInit.tsx

```diff
@@ -0,0 +1,21 @@
+import { PluginLoader, PluginStore, PluginStoreProvider } from '@openshift/dynamic-plugin-sdk';
+import * as React from 'react';
+import { initSharedScope, getSharedScope } from '../../shared-scope';
+
+const MinimalAppInit = React.lazy(async () => {
+  await initSharedScope();
```

Comment:
@karelhala This makes sense, we should move async application init logic outside the React render flow.

The idea here was to put any code related to plugin runtime infra initialization in one place (component).

Another idea is to keep `PluginLoader`'s `getSharedScope` function but type it as `() => Promise<AnyObject>` since shared scope init is the only async part here.

Anyway, since `PluginStore` creation should be done early during host application's startup and it does involve async operations, I'll just move it outside the React render flow in `packages/sample-app/src/app-minimal.tsx`.